runtime/src/main/scala/tailcall/runtime/service/HttpCache.scala

```diff
@@ -0,0 +1,56 @@
+package tailcall.runtime.service
+import tailcall.runtime.http.{HttpClient, Request}
+import zio.cache.{Cache, CacheStats, Lookup}
+import zio.http.Response
+import zio.{Duration, Exit, Task, ZIO, ZLayer}
+
+import java.text.SimpleDateFormat
+import java.time.Instant
+trait HttpCache  {
+  def get(key: Request): Task[Response]
+  def cacheStats: Task[CacheStats]
+}
+object HttpCache {
+  def ttl(res: Response, currentMillis: => Instant = Instant.now()): Option[Duration] = {
+    val headers      = res.headers.toList.map(x => String.valueOf(x.key).toLowerCase -> String.valueOf(x.value)).toMap
+    val cacheControl = headers.get("cache-control").map(_.split(",").map(_.trim).toSet).getOrElse(Set.empty)
+    val maxAge       = cacheControl.find(_.startsWith("max-age=")).map(_.split("=").last).flatMap(_.toLongOption)
+    val expires      = maxAge.map(_ => None).getOrElse(headers.get("expires"))
+    if (cacheControl.contains("private")) None
+    else if (expires.isEmpty) maxAge.map(Duration.fromSeconds)
+    else expires match {
+      case Some(value) =>
+        if (value matches "-1") { None }
+        else {
+          val date = new SimpleDateFormat("EEE, dd MMM yyyy HH:mm:ss z").parse(value).toInstant
```

Comment:
```suggestion
          val date = dataFormat.parse(value).toInstant
```
Move the format outside and reuse in test.